content/en/docs/contributing/style-guide.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Documentation style guide
+linkTitle: Style guide
+weight: 10
+cSpell:ignore: open-telemetry postgre style-guide textlintrc
+---
+
+We don't have an official style guide yet, but the current OpenTelemetry
+documentation style is inspired by the following style guides:
+
+- [Google Developer Documentation Style Guide](https://developers.google.com/style)
+- [Kubernetes Style Guide](https://kubernetes.io/docs/contribute/style/style-guide/)
+
+The following sections contain guidance that is specific to the OpenTelemetry
+project.
+
+## OpenTelemetry.io word list
+
+A list of OpenTelemetry-specific terms and words to be used consistently across
+the site.
+
+<!-- prettier-ignore-start -->
+| Term | Usage |
+| --- | --- |
+| OpenTelemetry | OpenTelemetry should always be capitalized. Don't use Open-Telemetry. |
+| OTel | OTel is the accepted short form of OpenTelemetry. Don't use OTEL. |
+| Collector | When referring to the OpenTelemetry Collector, always capitalize Collector. |
+| Repository | Code repository, lowercase when in the middle of a sentence. Don't use "repo" or "repos". |
+| OTEP | OpenTelemetry Enhancement Proposal. Write "OTEPs" as plural form. Don't write "OTep" or "otep". |
+| OpAMP | Open Agent Management Protocol. Don't write "OPAMP" or "opamp" in descriptions or instructions. |
+| OTLP | OpenTelemetry Protocol. Don't write "OTlP" or "otlp" in descriptions or instructions. |
+<!-- prettier-ignore-end -->
```

Comment:
A comment for later: I don't think that we should duplicate here in the form of a table any checks that the tools do. IMHO, we should just mention the tools and how to invoke them so that contributors know if they're using the proper spelling and capitalization.

Reply:
I think there's an educational, reference value in having our terminology presented like this. This is not too different from what style guides do with their wordlists:

* https://developers.google.com/style/word-list
* https://docs.splunk.com/Documentation/StyleGuide/current/StyleGuide/Usagedictionary
* https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/m/microsoft-store

Having a place that explain, for humans, what the proper capitalization or spelling is for OTel terminology also has the potential of ending discussions and preventing headaches during editing.

Reply:
Thanks for the links. I agree that it can make sense for both the terms OpenTelemetry and OTel, but not e.g. for Repository or simple cases of all caps for acronyms. But we can discuss these finer points later.

Also, if we can automate the creation of such a table somehow, that would satisfy my need to keep things [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

Reply:
Agree on the generic terminology! Automation would be great, too.

In general, while deeply imperfect, I see this initial PR as the foundations for deeper discussions around our contribution guidelines and style. We'll get to a great point soon, I'm sure!

Reply:
See https://github.com/open-telemetry/opentelemetry.io/pull/3846